### PR TITLE
add regtest support

### PR DESCRIPTION
**Pull request by fivepiece**
See #4241 for meta discussion

1. Add a `--regtest` option
2. Change the human readable part for bech32 addresses from `tb` to `bcrt`
3. Set an empty checkpoints parameter: `CHECKPOINTS = []`
4. Set the genesis block hash to `0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206`
5. Create a new `servers_regtest.json` only `localhost` as the server and use it for setting `DEFAULT_SERVERS`
6. Make sure that `--regtest` and `--testnet` are exclusive

### CODE REVIEW

File: lib/servers_regtest.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "127.0.0.1": {
+        "pruning": "-",
+        "s": "50002",
+        "t": "50001",
```
Comment (SomberNight) [inline lib/servers_regtest.json]: I guess these should rather be `'t': '51001', 's': '51002'`, which is also what you specified as `BitcoinRegtest.DEFAULT_PORTS`
https://github.com/kyuupichan/electrumx/blob/fbf0ec4d9b9bb5c4c6bdcbf27c3fd4d9def91670/lib/coins.py#L571

Comment (fivepiece) [reply lib/servers_regtest.json]: Yep, changing now.

**Comment by fivepiece** [comment]
thanks.  ping me when you want me to squash the commits

File: lib/constants.py [hunk 2]
```diff
@@ -93,6 +93,34 @@ class BitcoinTestnet:
     }
 
 
+class BitcoinRegtest:
```
Comment (SomberNight) [inline lib/constants.py]: Actually, I'm thinking it might be better to inherit from `BitcoinTestnet` so some of the fields don't need to be duplicated :P

Comment (fivepiece) [reply lib/constants.py]: No problem :)

**Comment by fivepiece** [comment]
Updated to address @SomberNight comments

**Pull request merged**